src/openms/source/DATASTRUCTURES/String.cpp

```diff
@@ -526,13 +528,11 @@ namespace OpenMS
     this->append(s);
     return *this;
   }
+  
+  ::size_t hash_value(String const& s)
+  {
+    boost::hash<std::string> hasher;
```

Comment:
Is this costly?

Reply:
dont know. but this is recommended from boost. https://www.boost.org/doc/libs/1_62_0/doc/html/hash/custom.html

Reply:
it has nothing but the operator()

Reply:
which returns hash_value. It is probably much safer to use it this way instead of trying to save one layer of abstraction.